core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/WindowsShareCopy.java

```diff
@@ -234,7 +237,8 @@ protected void terminated() {
 
     @Description("Specifies the number of threads to execute the copy operation")
     @Nullable
-    private final Integer numThreads;
+    @Macro
+    private Integer numThreads;
```

Comment:
The user should not have to publish a new pipeline every time a different buffer size or thread count is needed. If these fields are macro that allows the user to take these configurations as runtime arguments